lib/autoupdate/agent/process.go

```diff
@@ -466,6 +479,34 @@ func hasSystemD() (bool, error) {
 	return true, nil
 }
 
+// hasSystemDBelow returns true the version of systemd can be determined, and it
+// is below the provided version.
+func hasSystemDBelow(ctx context.Context, i int) bool {
+	cmd := exec.CommandContext(ctx, "systemctl", "--version")
+	out, err := cmd.Output()
+	if err != nil {
+		return false
+	}
+	if v, ok := parseSystemDVersion(out); ok && v < i {
+		return true
+	}
+	return false
+}
+
+// parseSystemDVersion parses the SystemD version from systemctl command output.
+func parseSystemDVersion(out []byte) (int, bool) {
+	first, _, _ := strings.Cut(string(out), "\n")
```

Comment:
Would this function work just as well if you omit the first line here completely? You'd split on the entire output and not just the first line, but I think everything would still be where you expect it to be.

Reply:
This wouldn't work for older systemd, unfortunately:
```
[root@bef448ecc578 /]# systemctl --version
systemd 219
+PAM +AUDIT +SELINUX +IMA -APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 -SECCOMP +BLKID +ELFUTILS +KMOD +IDN
[root@bef448ecc578 /]# systemctl --version|cat -vte
systemd 219$
+PAM +AUDIT +SELINUX +IMA -APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 -SECCOMP +BLKID +ELFUTILS +KMOD +IDN$
```

Splitting on space would give "systemd", "219\n+PAM", ...